### PR TITLE
Adding JAVA sdk doc

### PR DESCRIPTION
**Pull request by agamkrbit**
Added documentation for JAVA sdk

### CODE REVIEW

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+<dependency>
+  <groupId>cloud.eppo</groupId>
+  <artifactId>eppo-server-sdk</artifactId>
+  <version>1.0.0</version>
+</dependency>
+```
+
+### 2. Define an Assignment Logger
+
+The SDK requires an assignment logger to be passed on initialization. The SDK invokes the logger to capture assignment data whenever a variation is assigned. The below code example shows how to integrate the SDK with [Segment](https://segment.com/docs/) for logging events. You could also use your own logging system; the only requirement is that the SDK receives a `logAssignment` function.
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:23]: > The below code example shows how to integrate the SDK with [Segment](https://segment.com/docs/) for logging events.

The code example below does not show how to use it with Segment. Please either remove that line or include the Segment code in the example.

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+The below code example shows how to initialize the SDK with the event logger from the previous section and your API key:
+
+```java
+
+package org.example;
+
+import com.eppo.sdk.EppoClient;
+import com.eppo.sdk.dto.EppoClientConfig;
+
+public class App
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:61]: To be concise, just show the SDK code parts without including  `public class App` or `public static void main`

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+### 4. Assign Experiment Variations
+
+Before using the SDK to assign a variation, make sure your experiment is setup as follows:
+1. The experiment must be configured to use Eppo's randomization:
+![use-eppo-randomization](../../../../../static/img/connecting-data/UseEpposRandomization.png)
+2. The experiment must be started **OR** the `subjectKey` passed to the SDK must be added to one of its variation allow lists
+![start-experiment](../../../../../static/img/connecting-data/StartExperiment.png)
+
+If the above conditions are not met, the SDK will return `""` as the assignment.
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:83]:
```suggestion
If the above conditions are not met, the SDK will return `Optional.empty()` as the assignment.
```

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+The experiment **Traffic Allocation** setting determines the percentage of subjects the SDK will assign to experiment variations. For example, if the traffic allocation is 25%, the SDK will assign a variation for 25% of subjects and no assignment for the remaining 75% (unless the subject is part of an allow list).
+
+The SDK requires two inputs to assign a variation:
+- `experimentKey` - this should be the same as the “Experiment Key” field of an Eppo experiment
+- `subjectKey` - the entity ID that is being experimented on, typically represented by a uuid.
+
+The below code example shows how to assign a subject to an experiment variation:
+
+```java
+System.out.println(eppoClient.getAssignment("<subject-id>", "<experiment-name>").get());
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:98]:
```suggestion
Optional<String> assignedVariation = eppoClient.getAssignment("<SUBJECT-KEY>", "<EXPERIMENT-KEY>");
```

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+The SDK requires an assignment logger to be passed on initialization. The SDK invokes the logger to capture assignment data whenever a variation is assigned. The below code example shows how to integrate the SDK with [Segment](https://segment.com/docs/) for logging events. You could also use your own logging system; the only requirement is that the SDK receives a `logAssignment` function.
+
+Define an implementation of the Eppo `IAssignmentLogger` interface. This interface has one function: `logAssignment`.
+
+```java
+from com.eppo.sdk.dto.IAssignmentLogger;
+from com.eppo.sdk.dto.AssignmentLogData;
+
+public AssignmentLoggerImpl implements IAssignmentLogger {
```
[the remark this answers is not in the document]
Comment (agamkrbit) [reply docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md]: Done!

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md (hunk 4 above)
Comment (agamkrbit) [reply docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md]: Done!

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md (hunk 3 above)
Comment (agamkrbit) [reply docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md]: Thanks for checking! fixed it!

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md (hunk 2 above)
Comment (agamkrbit) [reply docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md]: Removed the other code now only showing the initialisation code

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md (hunk 1 above)
Comment (agamkrbit) [reply docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md]: Fixed the statement. Removed the line regarding segment integration

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# JAVA
+
+Eppo's JAVA SDK is open source:
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:6]:
```suggestion
Eppo's Java SDK is open source:
```

Nit: don't capitalize JAVA

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+### 2. Define an Assignment Logger
+
+The SDK requires an assignment logger to be passed on initialization. The SDK invokes the logger to capture assignment data whenever a variation is assigned. The below code example shows how to create your own assignment logger.
+
+Define an implementation of the Eppo `IAssignmentLogger` interface. This interface has one function: `logAssignment`.
+
+```java
+from com.eppo.sdk.dto.IAssignmentLogger;
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:28]:
```suggestion
import com.eppo.sdk.dto.IAssignmentLogger;
```

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+from com.eppo.sdk.dto.AssignmentLogData;
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:29]:
```suggestion
import com.eppo.sdk.dto.AssignmentLogData;
```

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+The SDK will invoke the `logAssignment` function with an `event` object that contains the following fields:
+
+| Field | Description | Example |
+| --------- | ------- | ---------- |
+| `experiment` (string) | An Eppo experiment key | "recommendation_algo" |
+| `subject` (string) | An identifier of the subject or user assigned to the experiment variation | UUID |
+| `variation` (string) | The experiment variation the subject was assigned to | "control" |
+| `timestamp` (Date) | The time when the subject was assigned to the variation | 2021-06-22T17:35:12.000Z |
+| `subjectAttributes` (Map<String, String>) | A free-form map of metadata about the subject. These attributes are only logged if passed to the SDK assignment function | `{ "country": "US" }` |
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:46]:
```suggestion
| `subjectAttributes` (Map<String, EppoValue>) | A free-form map of metadata about the subject. These attributes are only logged if passed to the SDK assignment function | `Map.of("device", EppoValue.valueOf("iOS") |
```

File: docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# JAVA
```
Comment (ploomis14) [inline docs/prerequisites/feature-flagging/randomization-sdk/server-sdks/java.md:4]:
```suggestion
# Java
```

**Review by ploomis14 — APPROVED** [review]

**Pull request merged**